conformance/tests/httproute-reference-policy.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"golang.org/x/exp/slices"
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteReferencePolicy)
+}
+
+var HTTPRouteReferencePolicy = suite.ConformanceTest{
+	ShortName:   "HTTPRouteReferencePolicy",
+	Description: "A single HTTPRoute in the gateway-conformance-infra namespace, with a backendRef in the gateway-conformance-web-backend namespace, should attach to Gateway in the gateway-conformance-infra namespace",
+	Manifests:   []string{"tests/httproute-reference-policy.yaml"},
+	Test: func(t *testing.T, s *suite.ConformanceTestSuite) {
+		if !slices.Contains(s.ExtendedSupport, suite.SupportReferencePolicy) {
+			t.Skip("Skipping ReferencePolicy conformance test")
+		}
```

Comment:
Instead of adding this inside the `Test` func, maybe add a `Features` field to `suite.ConformanceTest` and update our suite.Run func to consider this when running tests.